js/freewheel.js

```diff
@@ -261,9 +263,39 @@ OO.Ads.manager(function(_, $) {
         ].join('')
       ];
 
-      // TODO: Implement a timeout here in case fw_onAdRequestComplete doesn't get called - PBI-731
+      // Set up function to log error if ad response is null after maximum duration allowed for ad request to respond
+      // has exceeded.
+      var adRequestTimeout = _.bind(function(){
+        if (!fwContext._adResponse) {
+          var error = "ad request timeout";
+          OO.log("FW: freewheel ad request timeout");
+          amc.raiseAdError("FW: An ad error has occurred. The error string reported was: " + error);
+          slotEndedCallbacks[adRequestType]();
+          delete slotEndedCallbacks[adRequestType];
+          _clearAdRequestTimeout();
+        }
+      }, this);
+
       fwContext.submitRequest();
       fwAdDataRequested = true;
+      _setAdRequestTimeout(adRequestTimeout, amc.MAX_AD_REQUEST_TIMEOUT);
+    }, this);
+
+    /**
+     * Set timeout for ad request. Will call the provided callback
+     * if we timeout.
+     * @private
+     * @method Freewheel#_setAdRequestTimeout
+     * @param callback The function to call when we time out
+     * @param duration the time to wait before timing out
+     */
+    var _setAdRequestTimeout = _.bind(function(callback, duration){
+      if (adRequestTimeout) {
+        var error = "Ad Request Timeout already exists - bad state";
+        amc.raiseAdError("FW: An ad error has occurred. The error string reported was: " + error);
```

Comment:
You've used this in a couple of places. If you want, you can put this amc.raiseAdError into a function. That way you guarantee the format of the error message will always be the same.  Just a suggestion, not required.